### PR TITLE
Lettering: add font size filter

### PR DESCRIPTION
**Pull request by kaalleen**
Choosing a font for embroidery is a bit easier, when you can filter the font list for a specific target size.

### CODE REVIEW

File: fonts/aventurina/font.json [hunk 1]
```diff
@@ -8,6 +8,7 @@
     "default_glyph": "�",
     "min_scale": 0.6,
     "max_scale": 1.5,
+    "scale": 20,
```
Comment (george-steel) [inline fonts/aventurina/font.json:11]: this should be `size`

Comment (kaalleen) [reply fonts/aventurina/font.json]: Ops, thanks!!

Comment (george-steel) [inline fonts/aventurina/font.json]: should be "size"

File: fonts/geneva_simple/font.json [hunk 2]
```diff
@@ -7,9 +7,10 @@
     "default_glyph": "�",
     "min_scale": 0.7,
     "max_scale": 2.0,
+    "scale": 8,
```
Comment (george-steel) [inline fonts/geneva_simple/font.json]: same here

**Comment by george-steel** [comment]
I would recommend a rebase as the diff algorithm has gotten confused and has flagged a lot of unchanged lines. Other than that and the `scale` parameters that need changing to `size`, LGTM.

**Comment by kaalleen** [comment]
No, it's not confused. This is caused by unix style line breaks vs windows style line breaks.
Thanks for spotting down the scale / size issue.

File: lib/extensions/lettering.py [hunk 3]
```diff
@@ -52,12 +45,19 @@ def __init__(self, *args, **kwargs):
...
 
         self.font_chooser = wx.adv.BitmapComboBox(self, wx.ID_ANY, style=wx.CB_READONLY | wx.CB_SORT)
-        self.set_font_list()
         self.font_chooser.Bind(wx.EVT_COMBOBOX, self.on_font_changed)
 
+        self.font_filter = fs.FloatSpin(self, min_val=0, max_val=None, increment=1, value="0")
+        self.font_filter.SetFormat("%f")
+        self.font_filter.SetDigits(2)
+        self.font_filter.Bind(fs.EVT_FLOATSPIN, self.on_filter_changed)
+        self.font_filter.SetToolTip(_("Font size filter (mm)"))
```
Comment (george-steel) [inline lib/extensions/lettering.py:56]: It would be clearer as "Font size filter (mm), 0 for all sizes."

**Comment by george-steel** [comment]
It would be more intuitive id the scale box could be pre-set to get the filtered size when a font is selected

**Comment by george-steel** [comment]
I just tried this out and it's quite useful. auto-fillinf the scale field would definitely make it even better as most fonts do not show their sizes in their descriptions (which is needed to calculate it manually).

File: lib/extensions/lettering.py (hunk 3 above)
Comment (kaalleen) [reply lib/extensions/lettering.py]: Yes, right. Thanks!

**Comment by kaalleen** [comment]
> It would be more intuitive id the scale box could be pre-set to get the filtered size when a font is selected

Do you mean to update the filter value when a font is selected? I wouldn't like that, it could be pretty annoyin, when you are looking for a font of a specific size, you would need to set the value again after every tested font.

Or do you mean to add a info box with the size data for each font. Usually the fonts do have the size in their description (except for 2 fonts). But I could imagine, that it could be helfpul to output the size of a selected font (with the actual scale factor included). So users can see right away how big the final text is going to be.

**Comment by george-steel** [comment]
What I meant was to initialize the selected font's scale box to match the value selected in the filter instead of always initializing it at the minimum. Showing the original and calculated sizes alongside the description would be useful as well.

**Comment by kaalleen** [comment]
Oh, I see. That makes sense. I totally confused the filter and scale settings. Sorry.

**Pull request merged**